apps/hash-graph/lib/graph/src/shared/identifier/time.rs

```diff
@@ -1,23 +1,26 @@
 mod axis;
 mod bound;
 mod interval;
-mod projection;
+mod temporal_axes;
 mod timestamp;
 
 pub use self::{
-    axis::{DecisionTime, ProjectedTime, TemporalTagged, TimeAxis, TransactionTime},
+    axis::{DecisionTime, TemporalTagged, TimeAxis, TransactionTime, VariableAxis},
     bound::{
-        IncludedTimeIntervalBound, LimitedTimeIntervalBound, TimeIntervalBound,
-        UnboundedOrExcludedTimeIntervalBound,
+        InclusiveTemporalBound, LimitedTemporalBound, TemporalBound,
+        UnboundedOrExclusiveTemporalBound,
     },
-    interval::UnresolvedTimeInterval,
-    projection::{
-        DecisionTimeImage, DecisionTimeKernel, DecisionTimeProjection, Image, Kernel,
-        TimeProjection, TransactionTimeImage, TransactionTimeKernel, TransactionTimeProjection,
-        UnresolvedDecisionTimeImage, UnresolvedDecisionTimeKernel,
-        UnresolvedDecisionTimeProjection, UnresolvedImage, UnresolvedKernel, UnresolvedProjection,
-        UnresolvedTimeProjection, UnresolvedTransactionTimeImage, UnresolvedTransactionTimeKernel,
-        UnresolvedTransactionTimeProjection,
+    interval::UnresolvedTemporalInterval,
+    temporal_axes::{
+        PinnedTemporalAxis, TemporalAxes, UnresolvedPinnedTemporalAxis, UnresolvedTemporalAxes,
+        UnresolvedVariableTemporalAxis, VariableTemporalAxis,
     },
     timestamp::Timestamp,
 };
+use crate::interval::Interval;
+
+pub type TemporalInterval<A> = Interval<Timestamp<A>, TemporalBound<A>, TemporalBound<A>>;
+pub type LimitedTemporalInterval<A> =
```

Comment:
This naming seems a bit strange to me. The lower bound isn't necessarily timestamp limited?

Reply:
See below